subprojects/core/src/main/java/org/gradle/api/internal/TaskInternal.java

```diff
@@ -109,4 +110,12 @@
      */
     @Internal
     List<? extends ResourceLock> getSharedResources();
+
+    /**
+     * "Lifecycle dependencies" are dependencies of this task declared via an explicit {@link Task#dependsOn(Object...)} call,
+     * as opposed to the recommended approach of connecting producer tasks' outputs to consumer tasks' inputs.
+     * @return the dependencies of this task declared via an explicit {@link Task#dependsOn(Object...)}
```

Comment:
I think the other meaningful difference here is that a "lifecycle dependency" says that this task depends on the outcome of the other task, so we cannot execute this task if the other task fails.